File: examples/injectable-twitter-server/java/src/test/java/com/twitter/finatra/example/ExampleTwitterServerStartupTest.java
package com.twitter.finatra.example;

import java.util.Collections;

import com.google.inject.Stage;

import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

import com.twitter.inject.server.EmbeddedTwitterServer;

public class ExampleTwitterServerStartupTest extends Assert {

  private static final EmbeddedTwitterServer SERVER =
      new EmbeddedTwitterServer(
          new ExampleTwitterServer(),
          Collections.emptyMap(),
          Stage.PRODUCTION
      );

  @BeforeClass
  public static void setup() {
    SERVER.start();
  }

  @AfterClass
  public static void tearDown() {
    SERVER.close();
  }

  @Test
  public void startupTest() throws Exception {
    SERVER.assertHealthy(true);
  }
}

File: examples/injectable-twitter-server/scala/src/test/scala/com/twitter/finatra/example/ExampleTwitterServerStartupTest.scala
package com.twitter.finatra.example

import com.google.inject.Stage
import com.twitter.inject.server.{EmbeddedTwitterServer, FeatureTest}

class ExampleTwitterServerStartupTest extends FeatureTest {

  val server: EmbeddedTwitterServer = new EmbeddedTwitterServer(
    twitterServer = new ExampleTwitterServer,
    disableTestLogging = true,
    stage = Stage.PRODUCTION
  )

  test("Server#startup") {
    server.assertHealthy()
  }
}
